### PR TITLE
Vale

### PR DESCRIPTION
**Pull request by apinnick**
* [x] I am okay with my commits getting squashed when you merge this PR.
* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.11/Katello 4.13
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

**Comment by apinnick** [comment]
I have a feeling this will not work because I forked the repo. See the Vale GitHub action [documentation](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-github-action/):

> WARNING: Because of GitHub token permissions, the Vale GitHub Action cannot post annotations to pull requests from a forked repository.

**Comment by apinnick** [comment]
Looks like we might have to use a script instead. See [documentation](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-with-a-continuous-integration-service/).

**Comment by Lennonka** [comment]
> Looks like we might have to use a script instead. See [documentation](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-with-a-continuous-integration-service/).

I don't know about that. Possibly. Perhaps we should also turn it into a build target in the [Makefile](https://github.com/theforeman/foreman-documentation/blob/master/guides/Makefile)?

Otherwise, I hope someone more knowledgeable would help. @ekohl @evgeni ?

**Comment by evgeni** [comment]
I think I need more context to understand what's going on here.

**Comment by Lennonka** [comment]
@evgeni We want to add Vale, which is a linter for the RH style, to our GitHub PR checks.
Not sure how. Found [Vale as GitHub action](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-github-action/) in the Vale docs.

File: .github/workflows/lint-with-vale.yml
Comment (ekohl) [inline .github/workflows/lint-with-vale.yml:21]: Please ensure line endings in files

**Review by ekohl — COMMENTED** [review]
GitHub only picks up new actions from inside the repository. One workaround I've found is to temporarily push your branch to upstream. Once the action is known, you can delete your branch and continue working from your fork.

File: .vale.ini
Comment (evgeni) [inline .vale.ini:22]: This URL goes to a domain grabber. Did you mean https://vale.sh/docs/topics/scoping/?

**Comment by evgeni** [comment]
> GitHub only picks up new actions from inside the repository. One workaround I've found is to temporarily push your branch to upstream. Once the action is known, you can delete your branch and continue working from your fork.

That's not true for new workflows. Only when the repo didn't have actions at all before.

See https://github.com/theforeman/foreman-documentation/pull/3147/

File: .github/workflows/lint-with-vale.yml
Comment (evgeni) [inline .github/workflows/lint-with-vale.yml:26]: you can skip the env, the action fetches `secrets.GITHUB_TOKEN` itself, and you only need to override it if you want to give it a different token.

**Comment by apinnick** [comment]
It seems to work now. The check 'failed' after 1m and the annotations show the 2 errors I would expect.

File: .github/workflows/lint-with-vale.yml
Comment (Lennonka) [inline .github/workflows/lint-with-vale.yml:2]:
```suggestion
name: linter
```

File: .github/workflows/lint-with-vale.yml
Comment (Lennonka) [inline .github/workflows/lint-with-vale.yml:11]:
```suggestion
    name: lint-with-vale
```

**Review by Lennonka — CHANGES_REQUESTED** [review]
Can we give it names consistent with other checks, please?

File: .github/workflows/lint-with-vale.yml
Comment (apinnick) [reply .github/workflows/lint-with-vale.yml:2]: I do not recommend this because it is possible to set up multiple linters. Specific job names make debugging easier.

File: .github/workflows/lint-with-vale.yml
Comment (apinnick) [reply .github/workflows/lint-with-vale.yml:11]: Not sure why the name should be changed.

File: .github/workflows/lint-with-vale.yml
Comment (ekohl) [reply .github/workflows/lint-with-vale.yml:2]: Right now the check shows up as `Linting with Vale on pull request / Linting with Vale (pull_request)`. That's composed of:
* `Linting with Vale on pull request` (this line)
* `Linting with Vale` (name on line 11)
* `pull_request` (trigger, line 4)

You can see there's a lot of  duplication.

I'd still consider simplifying to:
```suggestion
name: Vale
```

And then make line 11 simply `Lint`. The result will be `Vale / Lint (pull_request)`. On the checks tab all jobs will be grouped under `Vale` as well (though there is only 1 job now).

You can also consider renaming the file to `vale.yml`. IMHO the `lint-with` is redundant and you can simply make the whole file just for all actions that Vale can do.

Comment (apinnick) [reply .github/workflows/lint-with-vale.yml]: Done

File: .github/workflows/vale.yml
Comment (ekohl) [inline .github/workflows/vale.yml:20]: Don't we actually want the check to fail if there's an error? Otherwise they're just warnings, right?

Comment (apinnick) [reply .github/workflows/vale.yml:20]: "fail_on_error: true" blocks merge. I don't recommend it until this feature has been thoroughly tested. [Documentation](https://redhat-documentation.github.io/vale-at-red-hat/docs/main/user-guide/using-vale-github-action/)

Comment (apinnick) [reply .github/workflows/vale.yml:20]: Yes, it's true that they are just warnings, but if you make a Vale pass mandatory, a merge might be blocked until someone can add an exception.

Comment (apinnick) [reply .github/workflows/vale.yml:20]: Or we could make it mandatory and see how it goes....

**Comment by maximiliankolb** [comment]
For the first month, I'd suggest to **not** block PRs if Vale fails. After all maintainers and regular contributors have some experience, I am open/in favor of making the check mandatory.

File: .github/workflows/vale.yml
Comment (apinnick) [reply .github/workflows/vale.yml:20]: That's what I thought. :-) So I will resolve this comment.

Comment (apinnick) [reply .github/workflows/vale.yml:20]: > A failing check shows up, but doesn't block merge unless you make it a required check. Just as the linkchecker today doesn't block merges but does clearly show.

I will add an error just to make sure that the Vale documentation is accurate.

**Review by Lennonka — APPROVED** [review]
Looks great!

**Review by maximiliankolb — APPROVED** [review]
Thanks Avital!

**Comment by Lennonka** [comment]
Regarding the naming. If you look at the checks, you can see that the "check class" goes first and then goes the specific "check name". For Vale, it's the other way around. But it's not a blocker.
image

File: .github/workflows/vale.yml
Comment (ekohl) [inline .github/workflows/vale.yml:6]: Is this recursive? I'd be worried this is globbing just on `guides/common/*.adoc` instead of `guides/common/**/*.adoc`.

Comment (apinnick) [reply .github/workflows/vale.yml:6]: It found errors in a guilds/common/module/<file>.

**Comment by apinnick** [comment]
> Regarding the naming. If you look at the checks, you can see that the "check class" goes first and then goes the specific "check name". For Vale, it's the other way around. But it's not a blocker. 

@Lennonka No, it doesn't work this way. If you look at the other workflow files, "deploy" or "lnkchecker" is the name of the YAML file and "check-html", etc., is the name of the job. The first element of the name does not refer to a 'class' or 'stage'. I could change it to "Vale / Vale" but I'm not sure how that would be any better.

**Review by ekohl — APPROVED** [review]
I think the current name is correct. It's indeed `Workflow / Job (Trigger)` and to me this is the best option.

File: .github/workflows/vale.yml
Comment (ekohl) [reply .github/workflows/vale.yml:6]: I think this was actually a problem. We can see in https://github.com/theforeman/foreman-documentation/pull/3165 that it's not triggered.

This PR also modified the other trigger (`vale.yml`) so it wasn't a problem (and vale doesn't care about it).

Looking at https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-pull_request-workflow-based-on-files-changed-in-a-pull-request they explicitly use `**.js`.

**Pull request merged**
